lib/nucleus_rails/generator.rb

```diff
@@ -0,0 +1,343 @@
+
+            return #{refs.variable_name.pluralize}.map do |#{refs.variable_name}|
+              #{refs.module_name}::View.new(#{refs.variable_name})
+            end
+          end
+        end
+
+        def create
+          render_response do |req|
+            policy = #{refs.class_name}Policy.new(current_user, #{refs.variable_name})
+            policy.enforce!(:can_write_#{refs.variable_name}?)
+
+            result = #{refs.module_name}::Update.call(update_params)
+
+            return result if !result.success?
+
+            #{refs.variable_name} = result.#{refs.variable_name}
+            return #{refs.module_name}::View.new(result.#{refs.entity_name})
+          end
+        end
+
+        def update
+          render_response do |req|
+            #{refs.variable_name} = #{class_name}Repository.find!(update_params[:id])
+            policy = #{refs.class_name}Policy.new(current_user, #{refs.variable_name})
+            policy.enforce!(:can_write_#{refs.variable_name}?, :can_access_#{refs.variable_name}?)
+
+            result = #{refs.module_name}::Create.call(create_params)
+
+            return result if !result.success?
+
+            #{refs.variable_name} = result.#{refs.variable_name}
+            return #{refs.module_name}::View.new(result.#{refs.entity_name})
+          end
+        end
+
+        def destroy
+          render_response do |req|
+            #{refs.variable_name} = #{class_name}Repository.find!(show_params[:id])
+            policy = #{refs.class_name}Policy.new(current_user, #{refs.variable_name})
+            policy.enforce!(:can_write_#{refs.variable_name}?, :can_access_#{refs.variable_name}?)
+
+            return #{refs.module_name}::Destroy.call(show_params)
+          end
+        end
+
+        private
+
+        def show_params
+          params.permit(#{refs.primary_key.to_sym})
+        end
+
+        def index_params
+          params.permit(*#{refs.columns.to_set.map(&:to_sym)})
+        end
+
+        def create_params
+          params.permit(*#{refs.columns.to_set.delete(:id).map(&:to_sym)})
+        end
+
+        def update_params
+          params.permit(*#{refs.columns.to_set.map(&:to_sym)})
+        end
+      end
+    RUBY
+
+    create_file("app/controllers/#{refs.folder_name}_controller.rb", content)
+  end
+
+  def update_routes(refs)
+    append_to_file("config/routes.rb", after: /Rails.application.routes.draw do/) do
+      <<-RUBY
+        resources :#{refs.folder_name}, only: [:show, :index, :create, :update, :destroy]
+      RUBY
+    end
+  end
+
+  def entity_details(table_name)
+    columns = ActiveRecord::Base.connection.columns(table_name).map(&:name)
+    entity_name = table_name.tableize.singularize
+    
+    OpenStruct.new(
+      table_name: table_name,                      # posts
+      primary_key: klass.primary_key,              # id
+      columns: columns,                            # ['id', 'title', 'body', etc...]
```

Comment:
add indexes and foreign keys. Add this object to a lookup hash to evaluate dependent relationships